tests/unittests/libretroshare/gxs/data_service/rsdataservice_test.cc

```diff
@@ -35,55 +36,47 @@ void test_groupStoreAndRetrieve(){
     setUp();
 
     int nGrp = rand()%32;
-    std::map<RsNxsGrp*, RsGxsGrpMetaData*> grps, grps_copy;
+	RsNxsGrpDataTemporaryList grps, grps_copy;
     RsNxsGrp* grp;
     RsGxsGrpMetaData* grpMeta;
-    for(int i = 0; i < nGrp; i++){
-        std::pair<RsNxsGrp*, RsGxsGrpMetaData*> p;
-       grp = new RsNxsGrp(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
-       grpMeta = new RsGxsGrpMetaData();
-       p.first = grp;
-       p.second = grpMeta;
-       init_item(*grp);
-       init_item(grpMeta);
-       grpMeta->mGroupId = grp->grpId;
-       grps.insert(p);
-       RsNxsGrp* grp_copy = new RsNxsGrp(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
-       *grp_copy = *grp;
-       RsGxsGrpMetaData* grpMeta_copy = new RsGxsGrpMetaData();
-       *grpMeta_copy = *grpMeta;
-       grps_copy.insert(std::make_pair(grp_copy, grpMeta_copy ));
-       grpMeta = NULL;
-       grp = NULL;
-   }
+
+    for(int i = 0; i < nGrp; i++)
+	{
+		std::pair<RsNxsGrp*, RsGxsGrpMetaData*> p;
+		grp = new RsNxsGrp(RS_SERVICE_TYPE_PLUGIN_SIMPLE_FORUM);
+		grpMeta = new RsGxsGrpMetaData();
+
+		init_item(*grp);
+		init_item(grpMeta);
+
+		grpMeta->mGroupId = grp->grpId;
+		grp->metaData = grpMeta ;
+
+		grps.push_back(grp);
+	}
 
     dStore->storeGroup(grps);
 
-    //use copy, a grps are deleted in store
-    grps.clear();
-    grps = grps_copy;
+    RsNxsGrpDataTemporaryMap gR;
+    RsGxsGrpMetaTemporaryMap grpMetaR;
 
-    std::map<RsGxsGroupId, RsNxsGrp*> gR;
-    std::map<RsGxsGroupId, RsGxsGrpMetaData*> grpMetaR;
     dStore->retrieveNxsGrps(gR, false, false);
     dStore->retrieveGxsGrpMetaData(grpMetaR);
 
-    std::map<RsNxsGrp*, RsGxsGrpMetaData*>::iterator mit = grps.begin();
-
     bool grpMatch = true, grpMetaMatch = true;
 
-    for(; mit != grps.end(); mit++)
+    for( std::list<RsNxsGrp*>::iterator mit = grps.begin(); mit != grps.end(); mit++)
     {
-        const RsGxsGroupId grpId = mit->first->grpId;
+        const RsGxsGroupId grpId = (*mit)->metaData->mGroupId;
 
         // check if it exists
         if(gR.find(grpId) == gR.end()) {
             grpMatch = false;
             break;
         }
 
-        RsNxsGrp *l = mit->first,
-        *r = gR[grpId];
+        RsNxsGrp *l = (*mit);

```

Comment:
just a side effect of changing from (*mit).something, to (*mit). The () are not needed of course.